File: java/src/main/java/com/carepet/model/SensorDAO.java
package com.carepet.model;

import com.datastax.oss.driver.api.mapper.annotations.Dao;
import com.datastax.oss.driver.api.mapper.annotations.Insert;
import com.datastax.oss.driver.api.mapper.annotations.Select;
import com.datastax.oss.driver.api.mapper.annotations.Update;

import java.util.UUID;

@Dao
public interface SensorDAO {
    @Insert
    void create(Sensor sensor);

    @Update
    void update(Sensor sensor);

    @Select
    Sensor get(UUID pet, UUID id);
}

File: csharp/Model/SensorDAO.cs
using System;
using System.Collections.Generic;
using Cassandra;
using Cassandra.Mapping;

namespace CarePet.Model
{
    public class SensorDAO
    {
        private readonly IMapper _mapper;

        public SensorDAO(ISession session)
        {
            _mapper = new Cassandra.Mapping.Mapper(session);
        }

        public void Create(Sensor sensor)
        {
            _mapper.Insert(sensor);
        }

        public void Update(Sensor sensor)
        {
            _mapper.Update(sensor);
        }

        public Sensor Get(Guid petId, Guid sensorId)
        {
            return _mapper.FirstOrDefault<Sensor>(
                "WHERE pet_id = ? AND sensor_id = ?",
                petId,
                sensorId
            );
        }

        public IEnumerable<Sensor> FindByPet(Guid petId)
        {
            return _mapper.Fetch<Sensor>(
                "WHERE pet_id = ?",
                petId
            );
        }
    }
}
